### PR TITLE
chore: Merge with upstream

### PR DESCRIPTION
**Pull request by JFreegman**
Upstream is currently staying up-to-date with Zig master instead of a major release. This PR will fix the build for Zig master (2024-11-14), but I think we should stop updating the API until the next major release, and then stick with that.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/zig-toxcore-c/10)

### CODE REVIEW

**Comment by freylax** [comment]
Dear JFreegman,
I did some improvements for the c-toxcore-build-with-zig and c-toxcore-zig repos:
- using a specific Zig Version for CI (0.14.0 , 2024.10.0-mach)
- local node test works now on my box and CI too.
I would like that thees changes go in your PR.

File: build.zig.zon [hunk 1]
```diff
@@ -1,15 +1,17 @@
 .{
-    .name = "zig-toxcore-c",
+    .name = "c-toxcore-zig",
```
Comment (iphydf) [inline build.zig.zon:2]: Why the rename?

Comment (freylax) [reply build.zig.zon:2]: oh, the original repository name at nodecum is c-toxcore-zig...
Should I rename the original one? This could mess things up or?

**Comment by JFreegman** [comment]
> Dear JFreegman, I did some improvements for the c-toxcore-build-with-zig and c-toxcore-zig repos:
> 
>     * using a specific Zig Version for CI (0.14.0 , 2024.10.0-mach)
> 
>     * local node test works now on my box and CI too.
>       I would like that thees changes go in your PR.

Just out of curiosity, why don't the bindings simply link libtoxcore as a system lib?

**Comment by freylax** [comment]
Linking to libtoxcore as a system lib would be possible for sure. However then you would have to install (and maybe build) it yourself. Maybe we could have both options, to use a installed system libtoxcore or a static zig build libtoxcore with a tight coupling to the used revision and building of the lib.
I realized the last option because it gets you a smooth zig build experience, but I see that the other approach is feasible as well.

File: build.zig.zon (hunk 1 above)
Comment (freylax) [reply build.zig.zon:2]: I mean how would github behave if I rename nodecum/c-toxcore-zig which is the origin of the fork of TokTok/zig-toxcore-c.

**Pull request closed without merging**